internal/rootcoord/root_coord.go

```diff
@@ -855,24 +858,34 @@ func (c *Core) Register() error {
 
 func (c *Core) Init() error {
 	var initError error = nil
+	if c.kvBaseCreate == nil {
+		c.kvBaseCreate = func(root string) (kv.TxnKV, error) {
+			return etcdkv.NewEtcdKV(Params.EtcdEndpoints, root)
+		}
+	}
 	c.initOnce.Do(func() {
 		connectEtcdFn := func() error {
 			if c.etcdCli, initError = clientv3.New(clientv3.Config{Endpoints: Params.EtcdEndpoints, DialTimeout: 5 * time.Second}); initError != nil {
 				log.Error("RootCoord, Failed to new Etcd client", zap.Any("reason", initError))
 				return initError
 			}
-			var ms *metaSnapshot
-			ms, initError = newMetaSnapshot(c.etcdCli, Params.MetaRootPath, TimestampPrefix, 1024)
+			if c.kvBase, initError = c.kvBaseCreate(Params.KvRootPath); initError != nil {
+				log.Error("RootCoord, Failed to new EtcdKV", zap.Any("reason", initError))
+				return initError
+			}
+			var metaKV kv.TxnKV
+			metaKV, initError = c.kvBaseCreate(Params.MetaRootPath)
```

Comment:
Don't think we should start multiple etcd client here, it shoud be a singleton.
Metakv and snapshotkv just has different root paths.

Reply:
Agree. But kv package does not have a way to create a `EtcdKV` with existing etcdclient. Will have a PR fixing this,